crypto/src/main/scala/coop/rchain/crypto/hash/Blake2b512Random.scala

```diff
@@ -150,12 +151,12 @@ object Blake2b512Random {
         val result =
           new Blake2b512Random(Blake2b512Block(slice.size.toByte), ByteBuffer.allocate(128))
         squashedBuilder += result
-        slice.grouped(2).foreach { pair =>
-          pair(0).digest.finalizeInternal(pair(0).lastBlock.array(), 0, chainBlock, 0)
-          if (pair.size > 1) {
-            pair(1).digest.finalizeInternal(pair(1).lastBlock.array(), 0, chainBlock, 64)
-          } else {
-            BLANK_HASH.asReadOnlyBuffer().get(chainBlock, 64, 64)
+        slice.grouped(4).foreach { quad =>
+          for (i <- 0 until quad.size) {
+            quad(i).digest.finalizeInternal(quad(i).lastBlock.array(), 0, chainBlock, i * 32)
```

Comment:
So previously we were
- merging 2 * 128 byte lastBlock-s into one 128-byte chainBlock,
  by compressing each of the lastBlock-s into an 8-long = 64-byte array
  and appending each of the compressed results to the chainBlock

and now we are 
- merging 4 * 128 byte lastBlock-s into one 128-byte chainBlock,
  by compressing each of the lastBlock-s into an 8-long = 64-byte array
  and appending half of each of compressed results' bytes to the chainBlock

I guess this is OK, because the result of compression is a uniform hash? (is it?)

Reply:
Previously we were merging 2 * 64 byte outputs into one 128 byte input.
Now we merge 3 * 32 byte outputs into one 128 byte input.